app/controllers/WhatIsTheUTRVariationsController.scala

```diff
@@ -66,10 +69,44 @@ class WhatIsTheUTRVariationsController @Inject()(
           Future.successful(BadRequest(view(formWithErrors, mode, draftId, routes.WhatIsTheUTRVariationsController.onSubmit(mode, draftId)))),
 
         value => {
-          for {
+
+          (for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(WhatIsTheUTRVariationPage, value))
             _              <- registrationsRepository.set(updatedAnswers)
-          } yield Redirect(routes.TrustStatusController.status(draftId))
+            claim       <- trustsStore.get(request.internalId, value)
+          } yield claim) flatMap { claim =>
+
+            lazy val redirectTo = request.affinityGroup match {
+              case Agent => enrolmentStoreConnector.getAgentTrusts(value) map {
+                case NotClaimed => Redirect(routes.TrustNotClaimedController.onPageLoad(draftId))
+                case _ =>
+
+                  val agentEnrolled = request.enrolments.enrolments exists { enrolment =>
+                    enrolment.key equals config.serviceName
+                  }
```

Comment:
Is this correct?

An agent may have many enrolments and this doesn't check if the UTR in the session is on their enrolments. This checks if _any_ UTR is on their enrolments?